python/hsfs/client/rondb_rest_client.py

```diff
@@ -0,0 +1,224 @@
+        else:
+            raise ValueError(
+                "Use the refresh_rondb_connection method to reset the rondb_client_connection"
+            )
+
+        # Set base_url
+        scheme = "https" if self._current_config[self.USE_SSL] else "http"
+        self._base_url = furl(
+            f"{scheme}://{self._current_config[self.HOST]}:{self._current_config[self.PORT]}/{self._current_config[self.SERVER_API_VERSION]}"
+        )
+
+        assert self._rest_client is not None, "RonDB Rest Client failed to initialise."
+        assert (
+            self._auth is not None
+        ), "RonDB Rest Client Authentication failed to initialise. Check API Key."
+        assert (
+            self._base_url is not None
+        ), "RonDB Rest Client Base URL failed to initialise. Check host and port parameters."
+        assert (
+            self._current_config is not None
+        ), "RonDB Rest Client Configuration failed to initialise."
+
+    def _get_default_rondb_rest_client_config(self) -> dict[str, Any]:
+        default_config = self._get_default_rondb_rest_client_static_parameters_config()
+        default_config.update(
+            self._get_default_rondb_rest_client_dynamic_parameters_config()
+        )
+        return default_config
+
+    def _get_default_rondb_rest_client_static_parameters_config(self) -> dict[str, Any]:
+        return {
+            self.TIMEOUT: _DEFAULT_RONDB_REST_CLIENT_TIMEOUT_MS,
+            self.VERIFY_CERTS: _DEFAULT_RONDB_REST_CLIENT_VERIFY_CERTS,
+            self.USE_SSL: _DEFAULT_RONDB_REST_CLIENT_USE_SSL,
+            self.SSL_ASSERT_HOSTNAME: _DEFAULT_RONDB_REST_CLIENT_SSL_ASSERT_HOSTNAME,
+            self.SERVER_API_VERSION: _DEFAULT_RONDB_REST_CLIENT_SERVER_API_VERSION,
+            self.HTTP_AUTHORIZATION: _DEFAULT_RONDB_REST_CLIENT_HTTP_AUTHORIZATION,
+        }
+
+    def _get_default_rondb_rest_client_dynamic_parameters_config(
+        self,
+    ) -> dict[str, Any]:
+        url = furl(self._get_rondb_rest_server_endpoint())
+        return {
+            self.HOST: url.host,
+            self.PORT: url.port,
+            self.CA_CERTS: client.get_instance()._get_ca_chain_path(),
+        }
+
+    def _get_rondb_rest_server_endpoint(self) -> str:
+        if isinstance(client.get_instance(), client.external.Client):
+            external_domain = self.variable_api.get_loadbalancer_external_domain()
+            if external_domain == "":
+                external_domain = client.get_instance().host
+            return f"https://{external_domain}:{_DEFAULT_RONDB_REST_CLIENT_PORT}"
+        else:
+            service_discovery_domain = self.variable_api.get_service_discovery_domain()
+            if service_discovery_domain == "":
+                raise FeatureStoreException("Service discovery domain is not set.")
+            return f"https://rdrs.service.{service_discovery_domain}:{_DEFAULT_RONDB_REST_CLIENT_PORT}"
+
+    def _send_request(
```

Comment:
```suggestion
    def send_request(
```

should be a public method

Reply:
It is private in the Hopsworks base client, but fair enough I can make it public and document it appropriately